src/Hosting/Hosting/src/Internal/HostingApplicationDiagnostics.cs

```diff
@@ -430,28 +430,33 @@ private void RecordRequestStartMetrics(HttpContext httpContext)
             }
         }
 
+        // The trace id was successfully extracted, so we can set the trace state
+        // https://www.w3.org/TR/trace-context/#tracestate-header
         if (!string.IsNullOrEmpty(requestId))
         {
             if (!string.IsNullOrEmpty(traceState))
             {
                 activity.TraceStateString = traceState;
             }
-            var baggage = _propagator.ExtractBaggage(headers, static (object? carrier, string fieldName, out string? fieldValue, out IEnumerable<string>? fieldValues) =>
```

Comment:
Is there any chance that this was being skipped for performance reasons?  There should have been a comment, if that were the case, but it seems conceivable that we wanted to skip this work for activities not included in distributed tracing.

Reply:
I don't think so honestly. I think it was a misunderstanding and a link has been made between Trace Context and Baggage specifications even though they are independent of each other.